src/postprocess.ts

```diff
@@ -52,70 +41,96 @@ function createBabelTransformer(
+
+  // create a new object we'll use as options with the sliced presets and plugins
+  const optionsBase = { ...options, presets, plugins };
+
+  delete optionsBase.cacheDirectory;
+
+  const babelTransformer = (
+    codeSourcemapPair: jest.TransformedSource,
     filename: string,
-    config: JestConfig,
-    transformOptions: TransformOptions,
-  ): CodeSourceMapPair => {
-    const theseOptions = Object.assign(
-      { filename, inputSourceMap: codeSourcemapPair.map },
-      options,
-    );
+    config: jest.ProjectConfig,
+    transformOptions: JestCacheKeyOptions,
+  ): jest.TransformedSource => {
+    const inputSourceMap =
+      typeof codeSourcemapPair.map === 'string'
+        ? JSON.parse(codeSourcemapPair.map)
+        : codeSourcemapPair.map;
+    const theseOptions = { ...optionsBase, filename, inputSourceMap };
     if (transformOptions && transformOptions.instrument) {
       theseOptions.auxiliaryCommentBefore = ' istanbul ignore next ';
       // Copied from jest-runtime transform.js
-      theseOptions.plugins = theseOptions.plugins.concat([
-        [
-          istanbulPlugin,
-          {
-            // files outside `cwd` will not be instrumented
-            cwd: config.rootDir,
-            exclude: [],
-          },
-        ],
-      ]);
+      if (!hasIstanbul) {
+        theseOptions.plugins = [
+          ...theseOptions.plugins,
+          istanbulPluginConfig(config),
+        ];
+      }
     }
-    // Babel has incorrect typings, where the map is an object instead of a string. So we have to typecast it here
-    return (babel.transform(
+
+    // we typecast here because babel returns a more complete object than the one expected by jest
+    return babel.transform(
       codeSourcemapPair.code,
       theseOptions,
-    ) as any) as CodeSourceMapPair;
+    ) as jest.TransformedSource;
   };
+
+  return babelTransformer;
 }
 
 export const getPostProcessHook = (
-  tsCompilerOptions: CompilerOptions,
-  jestConfig: JestConfig,
-  tsJestConfig: TsJestConfig,
+  jestConfig: jest.ProjectConfig,
 ): PostProcessHook => {
+  const tsJestConfig = getTSJestConfig(jestConfig);
   if (tsJestConfig.skipBabel) {
     logOnce('Not using any postprocess hook.');
     // Identity function
     return input => input;
   }
 
-  const plugins = Array.from(
-    (tsJestConfig.babelConfig && tsJestConfig.babelConfig.plugins) || [],
-  );
-
+  const tsJestBabelConfig = tsJestConfig.babelConfig || {};
   const babelOptions: BabelTransformOptions = {
-    ...tsJestConfig.babelConfig,
+    ...tsJestBabelConfig,
     babelrc: tsJestConfig.useBabelrc || false,
-    plugins,
-    presets: tsJestConfig.babelConfig ? tsJestConfig.babelConfig.presets : [],
-    sourceMaps: tsJestConfig.disableSourceMapSupport !== true,
+    plugins: toArray(tsJestBabelConfig.plugins),
+    presets: toArray(tsJestBabelConfig.presets),
+    sourceMaps: tsJestConfig.disableSourceMapSupport ? false : 'both',
```

Comment:
They are not emitted, but with a `true` instead of a `both`, it we won't return all the data jest wants. It's something that has changed in latest babel, but is backward compat since `both` is truthy.

Reply:
Yeah but emitted I don't mean written to disk - i mean added as inline and returned so we can pass it to jest.